core-cube/src/main/java/org/apache/kylin/cube/model/CubeDescTiretreeGlobalDomainDictUtil.java

```diff
@@ -0,0 +1,164 @@
+import org.apache.kylin.metadata.model.TableDesc;
+import org.apache.kylin.metadata.model.TableRef;
+import org.apache.kylin.metadata.model.TblColRef;
+import org.apache.kylin.source.SourceManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+public class CubeDescTiretreeGlobalDomainDictUtil {
+    private static final Logger logger = LoggerFactory.getLogger(CubeDescTiretreeGlobalDomainDictUtil.class);
+
+    /**
+     * get reuse global tiretree global dic path
+     * @param tblColRef
+     * @param cubeDesc
+     * @return
+     */
+    public static String globalReuseDictPath(KylinConfig config, TblColRef tblColRef, CubeDesc cubeDesc) {
+        String globalResumeDictPath = null;
+        List<GlobalDict> globalDicts = cubeDesc.listDomainDict();
+        DataModelManager metadataManager = DataModelManager.getInstance(config);
+        CubeManager cubeManager = CubeManager.getInstance(KylinConfig.getInstanceFromEnv());
+        for (GlobalDict dict : globalDicts) {
+            if (dict.getSrc().getIdentity().equalsIgnoreCase(tblColRef.getIdentity())) {
+                String model = dict.getModel();
+                String cube = dict.getCube();
+                logger.info("cube:{} column:{} tiretree global domain dic reuse model:{} cube{} column:{} ", cubeDesc.getName() , tblColRef.getName(), model, cube, dict.getDesc());
+
+                DataModelDesc dataModel = metadataManager.getDataModelDesc(model);
+                if (Objects.isNull(dataModel)) {
+                    logger.error("get cube:{} column:{} tiretree global domain dic reuse DataModelDesc error", cubeDesc.getName(), tblColRef.getName());
+                    return null;
+                }
+
+                CubeInstance cubeInstance = cubeManager.getCube(cube);
+                CubeSegment cubeSegment = cubeInstance.getLatestReadySegment();
+
+                TblColRef colRef = dataModel.findColumn(dict.getDesc());
+                if(Objects.isNull(colRef)){
+                    logger.error("get cube:{} column:{} tiretree global domain dic TblColRef error");
+                    return null;
+                }
+
+                globalResumeDictPath = cubeSegment.getDictResPath(colRef);
+
+                if (StringUtils.isBlank(globalResumeDictPath)) {
+                    logger.error("get cube:{} column:{} tiretree global domain dic resume dict path error");
+                }
+                logger.error("get cube:{} column:{} tiretree global domain dic resume dict path is {}", globalResumeDictPath);
+                break;
+            }
+        }
+        return globalResumeDictPath;
+    }
+
+
+
+
+    /**
+     * add resuce global tiretree global dic for baseid job
+     * @param cubeDesc
+     * @param dumpList
+     */
+    public static void cuboidJob(CubeDesc cubeDesc, Set<String> dumpList) {
```

Comment:
The method name is not clear enough to clarify its function

Reply:
OK, I will fix it, thanks.